docs/src/content/docs/book/functions.mdx

```diff
@@ -193,10 +472,544 @@ contract ManualMethodId {
+
+```tact
+contract AntiPositive() {
+    // This getter's manually specified method ID
+    // will not collide with any autogenerated IDs
+    get(-42) fun negAns(): Int {
+        return -42;
+    }
+
+    get fun posAns(): Int {
+        return 42;
+    }
+}
+```
+
+## `init` — constructor function {#init}
+
+<Badge text="Contract scope" variant="note" size="medium"/><p/>
+
+The special constructor function `init(){:tact}` is run upon deployment of the contract. Unlike [contract parameters](/book/contracts#parameters), it performs a delayed initialization of the contract data, overriding the values of persistent state variables on-chain.
+
+```tact
+contract Example {
+    // Persistent state variables
+    var1: Int = 0; // initialized with default value 0
+    var2: Int;     // must be initialized in the init() function
+    var3: Int = 7; // initialized with default value 7
+
+    // Constructor function, which is run only once
+    init() {
+        self.var2 = 42;
+        self.var3 = 32; // overrides the default to 32
+    }
+}
+```
+
+For every receiver that doesn't alter the contract's variables, Tact optimizes away unnecessary storage overrides. However, contracts cannot benefit from such optimizations if the `init(){:tact}` function is present. That is because for contracts with `init(){:tact}` every receiver has to check whether the `init(){:tact}` function has run already and did that only once, and to do so, a special flag is implicitly stored in the contract's persistent state.
 
-Since method IDs are $19$-bit signed integers and some of them are reserved, only the inclusive ranges from $-2^{18}$ to $-5$ and from $2^{14}$ to $2^{18} - 1$ are free to be used by users. To avoid collisions, it is recommended to specify method IDs only in these ranges, avoiding the method IDs of Tact-specific getters mentioned above.
+Effectively, the behavior or `init(){:tact}` function can be simulated when using contract parameters instead — through adding a special [`Bool{:tact}`](/book/types#booleans) field and a function that would be used in place of the `init(){:tact}`. That function would be called at the beginning of every receiver, while the boolean field would be used as a flag.
+
+```tact
+contract CounterBuiltinInit {
+    // Persistent state variables
+    counter: Int as uint32;
+
+    // Constructor function, which is run only once
+    init() {
+        self.counter = 0;
+    }
+
+    receive() {
+        cashback(sender());
+    }
+
+    receive(_: Increment) {
+        self.counter += 1;
+    }
+
+    get fun counter(): Int {
+        return self.counter;
+    }
+}
+
+contract CounterDIYInit(
+    // Persistent state variables
+    initialized: Bool, // set this field to `false` during deployment
+    counter: Int as uint32, // set this field to `0` during deployment
+) {
+    // Internal function, which will be used in place of a pseudo-constructor function
+    fun init() {
```

Comment:
let's use some other name: this should also fix syntax highlighting and make it more obvious it's not a builtin construction